### ISSUE TITLE
array_map a sanitize function fails input sanitization validation

### ISSUE THREAD

**Issue by mehtryx** [WordPress-Coding-Standards/WordPress-Coding-Standards#339]
Hello,

I have what I thought must be something you've encountered in the past but after a good day of trying and searching I must admit I don't see any related issues.  Thus if I am mistaken, or better yet, incorrect in my assumptions or usage, please correct me!

We have the following snippet of code which fails with the message 'Detected usage of a non-sanitized input variable $_POST (WordPress.VIP.ValidatedSanitizedInput.InputNotSanitized)'  The line it is flagging is where we do the array_map to apply the sanitize_text_field function to each element of the $_POST['topic_order'] array.

The code itself looks like the following:

	/**
	 * Update each topic with a proper 'order'
	 * @return [type] [description]
	 */
	public function sort_topics_callback() {
		if ( isset( $_POST['topic_order'] ) ) {
			$topic_order = array_map( 'sanitize_text_field', $_POST['topic_order'] ); // input var ok
			for ( $i = 0; $i < count( $topic_order ); $i++ ) {
				$id_parts = split( '-', $topic_order[ $i ] );
				$id = intval( $id_parts[1] );
				$new_term_group = $i + 1;

				wp_update_term( $id, 'smrt-topic', array( 'term_group' => $new_term_group ) );
			}
		}

		die();
	}

**Comment by JDGrimes** [comment]
Hi @mehtryx, 

This actually has been mentioned before: https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/pull/294#issuecomment-68529542. That PR provides for whitelisting (and actually needs to be rewritten after #334), but it doesn't actually address `array_map()`.

**Comment by mehtryx** [comment]
Thanks, sorry for the duplicate then. Close this at your discretion and I'll keep eyes out for those others.

**Comment by JDGrimes** [comment]
> Thanks, sorry for the duplicate then. Close this at your discretion and I'll keep eyes out for those others.

Let's leave this open until support for `array_map()` is implemented, since it didn't have its own ticket yet.
